src/docdb/getDocumentClient.ts

```diff
@@ -3,13 +3,18 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as vscode from "vscode";
 import { DocumentClient } from "documentdb";
 import * as DocDBLib from 'documentdb/lib';
+import { ext } from "../extensionVariables";
 
 export function getDocumentClient(documentEndpoint: string, masterKey: string, isEmulator: boolean): DocumentClient {
     const documentBase = DocDBLib.DocumentBase;
     var connectionPolicy = new documentBase.ConnectionPolicy();
-    connectionPolicy.DisableSSLVerification = isEmulator;
+
+    let vscodeStrictSSL: boolean | undefined = vscode.workspace.getConfiguration().get<boolean>(ext.settingsKeys.vsCode.proxyStrictSSL);
+    let strictSSL = !isEmulator && vscodeStrictSSL;
+    connectionPolicy.DisableSSLVerification = !strictSSL;
```

Comment:
Any reason behind two boolean operator statements instead of `isEmulator || !vscodeStrictSSL` ?